File: admin_portal/src/app/enum/RoomStatus.ts
export class RoomStatus{

  readonly key: string;
  readonly value: string;

  constructor(key: string, value: string) {
      this.key = key;
      this.value = value;
  }

  public static Booked = new RoomStatus("Booked", "Booked");
  public static Occupied = new RoomStatus("Occupied", "Occupied");
  public static Unoccupied = new RoomStatus("Unoccupied", "Unoccupied");
  public static Ready = new RoomStatus("Ready", "Ready");
 
  static getAllEnumList(): Array<RoomStatus> {
    return [
      RoomStatus.Booked,
      RoomStatus.Occupied,
      RoomStatus.Unoccupied,
      RoomStatus.Unoccupied,
    ];
  }
}

File: ecommerce_api/src/main/java/com/example/ecommerce_api/enums/RoomStatus.java
package com.example.ecommerce_api.enums;

import com.example.ecommerce_api.exception.InvalidEnumValueException;

public enum RoomStatus {

  Booked("Booked","Booked"),
  Occupied("Occupied","Occupied"),
  Unoccupied("Unoccupied","Unoccupied"),
  Ready("Ready","Ready");

  private final String key;
  private final String value;

  RoomStatus(String key,String value) {
    this.key = key;
    this.value = value;
  }

  public String getKey() {
    return key;
  }
  public String getValue() {
    return value;
  }

  public static RoomStatus getEnumByValue(String value) {
    for (RoomStatus item : RoomStatus.values()) {
      if (item.getValue().equals(value)) {
        return item;
      }
    }
    throw new InvalidEnumValueException();
  }
}
